wordpress-mobile/WordPress-iOS
# Themes browsing on Jetpack sites

**Fixes** #7468 

On this PR we are adding support for Themes browsing (no activation or customization yet) on Jetpack sites. The feature is behind a develop feature flag since it doesn't make sense to allow browsing without customization and activation. 

For non Jetpack sites everything should work as it did before.
For Jetpack sites theme browsing is divided in two sections: Uploaded Themes and WP.com Themes as in Calypso. Each section has a header with the total number of themes:

![simulator screen shot aug 9 2017 13 45 07](https://user-images.githubusercontent.com/5558824/29133193-102c70f2-7d09-11e7-8b28-3775eee7973e.png) ![simulator screen shot aug 9 2017 13 45 21](https://user-images.githubusercontent.com/5558824/29133194-1063bce2-7d09-11e7-88c6-5e9bf2049309.png)

**To test:**

1. Make sure theme browsing still works correctly on non JP sites
2. Make sure theme browsing works correctly on JP sites
3. Make sure the section header (only added for JP sites) looks good on IPad + IPhone + RTL and that the number displayed (total number of themes, uploaded or WP.com) matches Calypso

Note: There's a list of issues on #7464 that have to be checked, I will do that as part of #7650 and # #7469, this PR only covers theme browsing for Jetpack sites. The active theme will not be correctly selected in the list when it is a WP.com theme because of some "Fun with IDs" (basically after a theme is installed its ID changes to ThemeID-wpcom and we fail to identify them as the same theme).

Needs review: @koke
